services/api/src/models/audit-log.js

```diff
@@ -0,0 +1,38 @@
+const mongoose = require('mongoose');
+const { createSchema } = require('../utils/schema');
+const { pick, isEmpty } = require('lodash');
+
+const definition = require('./definitions/audit-log.json');
+
+const schema = createSchema(definition.attributes);
+
+schema.statics.getFieldsContext = function (ctx) {
+  return {
+    user: ctx.state.authUser?.id,
+    requestMethod: ctx.request.method,
+    requestUrl: ctx.request.url,
+    routeNormalizedPath: ctx.routerPath,
+    routePrefix: ctx.router.opts.prefix,
+  };
+};
+
+schema.statics.getDiffObject = function getDiffObject(object, fields) {
+  const pathsModified = object.directModifiedPaths();
```

Comment:
@dominiek / @andrewplummer there is a somewhat of a problem with directModifiedPaths https://mongoosejs.com/docs/api/document.html#document_Document-directModifiedPaths. This works totally fine  ... but naturally doesn't work with auto population (because its not really a build in solution in mongoose). And I don't see an easy solution out of that. 

The best solution for me would be to separate the e.g. user object (auto populate) and the user id.
so we would have 

```
{
  userId: "ObjectId"
  user: {...populated  user object}
}
```

Reply:
#1 I'm not sure if the diff object is a requirement (I'm assuming it is to make things simpler from looking up multiple documents)
#2 I wouldn't really expect it to work with autopopulate anyway as an update is by definition not "deep" so I'm wondering if this is really even a problem? You just want to see which IDs changed no?

Reply:
yep only care about objectId, which is what doesn't work, because the auto-populate changes the value of from an objectId to a full populated object. 

still thinking of solutions ... let me know if you have ideas

Reply:
there is a `.depopulate` method I know... might be of help?

Reply:
I don't think we need this to work on populated objects.

Reply:
but we do want it to work in the e.g. if the objectId was changed right? (which also not work due to the auto populate replacing the objectId with the full populated object).

Reply:
It would be nice.

Reply:
Not too worried about it now